pike/auth.py

```diff
@@ -69,9 +76,9 @@ def __init__(self, conn, creds=None):
              self.context) = kerberos.authGSSClientInit(
                 "cifs/" + conn.server,
                 gssmech=2,
-                user=user,
-                password=password,
-                domain=domain)
+                user=user.encode("utf-8"),
```

Comment:
Is it worth the trouble to make the encode/decode version a global constant string, e.g.

    ENCODING = "utf-8"

elsewhere

     from constants import ENCODING    

Something like that.

I just see a lot of repetitive magic strings here.

Reply:
i think this is a good point, and also reminded me that i have _no idea_ what encoding this function is expecting those arguments to be in. this is the smb world, so a lot of stuff actually uses utf-16 little endian. i'll have to go check.

so, while normally i wouldn't bother with `utf-8` being a magic string, i think in this case, it's probably good to figure out what is right and then use it consistently.

Reply:
This is turning out to be a much bigger problem that i expected. I'm thinking that I'll just nerf gss NTLM auth. It's _really_ only support with the likewise gssapi layer, and building this on onefs and testing it is showing that it doesn't seem to work anymore.